routing/datasource_test.go

```diff
@@ -1,12 +1,16 @@
-package routing
+package routing_test
```

Comment:
To use exiting predicates test has to be in a different package. To test package private `processRouteDef` it is exported for tests similar to go stdlib, see `export_test.go`

Reply:
you can also add your new test in a separate `foo_test.go` file with the package change.

Reply:
There was already a test `TestErrorWhenUsingPrediateAsFilter` for error messages, to extend it I needed to have another predicate and filter. Separate `foo_test.go` with different package would require `export_test.go` anyway unless it would test different method.

Reply:
ok